core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -968,14 +968,14 @@ private PointList fetchWayGeometry_(long edgePointer, boolean reverse, int mode,
             geoRef += 4L;
             bytes = new byte[count * nodeAccess.getDimension() * 4];
             wayGeometry.getBytes(geoRef, bytes, bytes.length);
-        } else if (mode == 0)
+        } else if (mode == FetchWayGeometry.PILLAR_ONLY)
             return PointList.EMPTY;
 
-        PointList pillarNodes = new PointList(count + mode, nodeAccess.is3D());
+        PointList pillarNodes = new PointList(FetchWayGeometry.count(count, mode), nodeAccess.is3D());
         if (reverse) {
-            if ((mode & 2) != 0)
+            if (mode == FetchWayGeometry.ALL || mode == FetchWayGeometry.PILLAR_AND_ADJ)
                 pillarNodes.add(nodeAccess, adjNode);
-        } else if ((mode & 1) != 0)
+        } else if (mode == FetchWayGeometry.ALL || mode == FetchWayGeometry.BASE_AND_PILLAR)
             pillarNodes.add(nodeAccess, baseNode);
```

Comment:
This is a bit more lengthy now ;)

Reply:
IMO its more lengthy and a lot more readable. With the previous code I did not understand what happened here without looking up what e.g. mode=2 means.